src/ber/de.rs

```diff
@@ -267,13 +267,16 @@ impl<'input> crate::Decoder for Decoder<'input> {
                             *last &= !((1 << bits) - 1);
                         }
 
-                        let string = types::BitString::from_vec(buffer);
-
-                        if string.not_any() {
-                            Ok(types::BitString::new())
-                        } else {
-                            Ok(string)
-                        }
+                        let mut string = types::BitString::from_vec(buffer);
+                        // TODO:
+                        // Using `saturating_sub` here ensures that the `flip1` test passes,
+                        // but returning an error on an overflow is probably more correct,
+                        // at least for DER.
+                        // Need to consult the spec...
+                        let bit_length = string.len().saturating_sub(bits as usize);
+                        string.truncate(bit_length);
```

Comment:
I'd switch this to `checked_sub` and then return an error, I'm pretty sure the spec doesn't cover what you should do here

Reply:
I've taken a look at X.690 again:
The only way that we can get an overflow here is if `string.len() == 0 && bits > 0`, meaning the BIT STRING was empty but the unused bits octet was non-zero.
Thus clause 8.6.2.3 applies, and the right thing to do is error.
Will do this now...